tests/reloading_test.cpp

```diff
@@ -117,3 +111,157 @@ TEST_CASE( "reload_gun_with_swappable_magazine", "[reload],[gun]" )
+                CHECK( mag.ammo_remaining() > 0 );
+                CHECK( mag.contents.front().type == ammo.type );
+            }
+            WHEN( "the player triggers auto reload again" ) {
+                g->reload( false );
+                REQUIRE( dummy.activity );
+                process_activity( dummy );
+
+                THEN( "The magazine is loaded into the gun" ) {
+                    CHECK( dummy.weapon.ammo_remaining() > 0 );
+                }
+                WHEN( "the player triggers auto reload again" ) {
+                    g->reload( false );
+                    THEN( "No activity is generated" ) {
+                        CHECK( !dummy.activity );
+                    }
+                }
+            }
+        }
+        GIVEN( "the player also has an extended magazine" ) {
+            item &mag2 = dummy.i_add( item( "glockbigmag", 0, 0 ) );
+            const cata::optional<islot_magazine> &magazine_type2 = mag2.type->magazine;
+            REQUIRE( magazine_type2 );
+            REQUIRE( ammo_type->type.count( magazine_type2->type ) != 0 );
+            REQUIRE( mag2.ammo_remaining() == 0 );
+
+            WHEN( "the player triggers auto reload" ) {
+                g->reload( false );
+                REQUIRE( dummy.activity );
+                process_activity( dummy );
+
+                THEN( "the associated magazine is reloaded" ) {
+                    CHECK( mag.ammo_remaining() > 0 );
+                    CHECK( mag.contents.front().type == ammo.type );
+                }
+                WHEN( "the player triggers auto reload again" ) {
+                    g->reload( false );
+                    REQUIRE( dummy.activity );
+                    process_activity( dummy );
+
+                    THEN( "The magazine is loaded into the gun" ) {
+                        CHECK( dummy.weapon.ammo_remaining() > 0 );
+                    }
+                    WHEN( "the player triggers auto reload again" ) {
+                        g->reload( false );
+                        REQUIRE( dummy.activity );
+                        process_activity( dummy );
+
+                        THEN( "the second associated magazine is reloaded" ) {
+                            CHECK( mag2.ammo_remaining() > 0 );
+                            CHECK( mag2.contents.front().type == ammo.type );
+                        }
+                        WHEN( "the player triggers auto reload again" ) {
+                            g->reload( false );
+                            THEN( "No activity is generated" ) {
+                                CHECK( !dummy.activity );
+                            }
+                        }
+                    }
+                }
+            }
+            GIVEN( "the player also has another gun with ammo" ) {
+                item &mag2 = dummy.i_add( item( "glockbigmag", 0, 0 ) );
+                const cata::optional<islot_magazine> &magazine_type2 = mag2.type->magazine;
```

Comment:
(Un)surprisingly breaks tests in osxcross-builds:

https://ci.narc.ro/job/Cataclysm-Matrix/Graphics=Curses,Platform=OSX/8648/console

```
13:04:20 CCACHE_CPP2=1 ccache /usr/local/osxcross/osxcross/target/bin/x86_64-apple-darwin15-clang++ -DRELEASE -DMACOSX -DGIT_VERSION -DBACKTRACE -DLOCALIZE -DUSE_HOME_DIR -ffast-math -O0  -Wall -Wextra    -fsigned-char -stdlib=libc++ -Werror -std=c++11 -MMD -MP -mmacosx-version-min=10.7 -I/usr/local/osxcross/libs/gettext/include -I/usr/local/osxcross/libs/ncurses/include -I../src -Wno-unused-variable -Wno-sign-compare -Wno-unknown-pragmas -Wno-parentheses -MMD -MP -Wall -Wextra -c savegame_test.cpp -o obj/savegame_test.o
13:04:21 reloading_test.cpp:263:55: error: unused variable 'magazine_type2' [-Werror,-Wunused-variable]
13:04:21                 const cata::optional<islot_magazine> &magazine_type2 = mag2.type->magazine;
13:04:21                                                       ^
13:04:21 1 error generated.
```